groups/nts/ntsa/ntsa_tcpcongestioncontrol.h

```diff
@@ -0,0 +1,193 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_NTSA_TCPCONGESTIONCONTROL
+#define INCLUDED_NTSA_TCPCONGESTIONCONTROL
+
+#include <bsls_ident.h>
+BSLS_IDENT("$Id: $")
+
+#include <ntsa_error.h>
+#include <ntscfg_platform.h>
+#include <ntsscm_version.h>
+#include <bsl_array.h>
+#include <bsl_ostream.h>
+#include <bsl_string.h>
+#include <bsl_span.h>
+
+namespace BloombergLP {
+namespace ntsa {
+
+/// Provide a value identifying tcp congestion control algorithm.
+///
+/// @details
+/// Provide a value-semantic type that identifies tcp congestion control
+/// algorithm.
+///
+/// @par Attributes
+/// This class is composed of the following attributes.
+///
+/// @li d_buffer:
+/// Buffer which contains name of the tcp congestion control algorithm.
+///
+/// @par Thread Safety
+/// This class is not thread safe.
+///
+/// @ingroup module_ntsa_system
+
+class TcpCongestionControl
+{
+  public:
+    enum { TCP_CA_NAME_MAX = 16 };
+
+    enum Algorithm {
+        e_RENO = 0,
+        e_CUBIC,
+        e_BBR,
+        e_BIC,
+        e_DCTCP,
+        e_DIAG,
+        e_HIGHSPEED,
+        e_HTCP,
+        e_HYBLA,
+        e_ILLINOIS,
+        e_LP,
+        e_NV,
+        e_SCALABLE,
+        e_VEGAS,
+        e_VENO,
+        e_WESTWOOD,
+        e_YEAH
+    };
+
+  private:
+    enum { BUFFER_SIZE = TCP_CA_NAME_MAX + 1 };
+
+    bsl::array<char, BUFFER_SIZE> d_buffer;
+    bsl::uint8_t d_size; //including 0 terminator
+
+  public:
+    /// Create new TcpCongestionControl instance having the default value.
+    TcpCongestionControl();
+
+    /// Create new TcpCongestionControl instance having the same value as the
+    /// `specified` original.
+    TcpCongestionControl(const TcpCongestionControl& original);
+
+    /// Destroy this object.
+    ~TcpCongestionControl();
+
+    /// Assign the value of the specified 'other' object to this object.
+    /// Return a reference to this modifiable object.
+    TcpCongestionControl& operator=(const TcpCongestionControl& other);
+
+    /// Set algorithm name to the specified `value`. The value is expected to
+    /// be a null-terminated string not longer than TCP_CA_NAME_MAX. Return the
+    /// error.
+    ntsa::Error setAlgorithmName(const char* value);
+
+    /// Set algorithm name to the specified `value`. Length of the `value` is
+    /// expected to be not longer than TCP_CA_NAME_MAX. Return the error.
+    ntsa::Error setAlgorithmName(const bsl::string& value);
+
+    /// Return a span pointing to a null-terminated c-string containing name of
+    /// the tcp congestion control algorithm and it's size including '\0'
+    /// character
+    bsl::span<const char> getAlgorithm() const;
```

Comment:
Rename `getAlgorithm()` to simply `algorithm()`. The style of these value-semantic types has accessor functions typically named as nouns, unless their implementation does something significant, like performing a system call.